pytest_django/plugin.py

```diff
@@ -386,6 +386,20 @@ def _django_setup_unittest(request, django_db_blocker):
 
         cls = request.node.cls
 
+        def _cleaning_debug(self):
+            testMethod = getattr(self, self._testMethodName)
+            skipped = (
+                getattr(self.__class__, "__unittest_skip__", False) or
+                getattr(testMethod, "__unittest_skip__", False))
+
+            if not skipped:
+                self._pre_setup()
+            super(cls, self).debug()
+            if not skipped:
+                self._post_teardown()
+
+        cls.debug = _cleaning_debug
+
```

Comment:
Needs a comment/doc!

It was mentioned to report it for Django itself also.. has that been done?